docs/docs/guides/18_migration_viem/index.md

```diff
@@ -4,4 +4,261 @@ title: 'Migration from Web3.js to Viem'
+import { mainnet } from 'viem/chains'
+
+const client = createWalletClient({
+  chain: mainnet,
+  transport: custom(window.ethereum!)
+})
+```
+
+## Wallets and Accounts - Generate Private Key
+
+If your code generates private keys, here’s how to migrate that functionality. In web3.js if you are using:
+
+```javascript
+const privateKey = web3.eth.accounts.create().privateKey;
+console.log(privateKey);
+```
+
+To migrate this to Viem, you'll use the `generatePrivateKey()` function from the 'viem/accounts' module.
+
+```javascript
+import { generatePrivateKey } from 'viem/accounts';
+
+const privateKey = generatePrivateKey();
+console.log(privateKey);
+```
+
+## Wallets and Accounts - Create a Wallet
+
+When migrating from Web3.js to Viem, you'll need to update how you create and manage wallets. The process of adding accounts to wallets differs between the two libraries. In web3.js :
+
+```javascript
+const web3 = new Web3();
+const wallet = web3.eth.accounts.wallet.add(privateKey);
+console.log(wallet[0].address);
+```
+
+To migrate this to Viem, you'll use privateKeyToAccount() to create an account, and then can pass it to createWalletClient() for using it with client.
+
+```javascript
+import { createWalletClient, http } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+import { mainnet } from 'viem/chains';
+
+const account = privateKeyToAccount(privateKey);
+
+const client = createWalletClient({
+	account,
+	chain: mainnet,
+	transport: http(),
+});
+```
+
+## Signing Messages
+
+Update how you handle message signing, following is web3.js example:
+
+```javascript
+const signature = web3.eth.accounts.sign('Some data', privateKey).signature;
+console.log(signature);
+```
+
+To sign message using Viem, you can use `signMessage()` method.
+
+```javascript
+import { createWalletClient, custom } from 'viem'
+import { mainnet } from 'viem/chains'
+
+const walletClient = createWalletClient({
+  chain: mainnet,
+  transport: custom(window.ethereum!),
+});
+
+const [account] = await walletClient.getAddresses();
+
+const signature = await walletClient.signMessage({
+  account,
+  message: 'Some data',
+});
+
+```
+
+## Transaction
+
+When migrating transaction sending code, you'll need to update how transactions are signed and sent.
+
+```javascript
+const tx = await web3.eth.sendTransaction({
+	from: account,
+	to: '0x92d3267215Ec56542b985473E73C8417403B15ac',
+	value: web3.utils.toWei('0.001', 'ether'),
+});
+```
+
+In Viem there is `sendTransaction()` function avalible with walletClient.
+
+```javascript
+import { createWalletClient, custom } from 'viem'
+import { mainnet } from 'viem/chains'
+
+const walletClient = createWalletClient({
+  chain: mainnet,
+  transport: custom(window.ethereum!),
+});
+
+const [account] = await walletClient.getAddresses();
+
+const hash = await walletClient.sendTransaction({
+  account,
+  to: '0x92d3267215Ec56542b985473E73C8417403B15ac',
+  value: 100000000000000n
```

Comment:
would be good to use human friendly parseEther here, as in web3js snippet the value is parsed.